src/libImaging/AlphaComposite.c

```diff
@@ -25,13 +25,12 @@ ImagingAlphaComposite(Imaging imDst, Imaging imSrc) {
     int x, y;
 
     /* Check arguments */
-    if (!imDst || !imSrc || strcmp(imDst->mode, "RGBA") ||
-        imDst->type != IMAGING_TYPE_UINT8 || imDst->bands != 4) {
+    if (!imDst || !imSrc ||
+        (strcmp(imDst->mode, "RGBA") && strcmp(imDst->mode, "LA"))) {
         return ImagingError_ModeError();
     }
 
-    if (strcmp(imDst->mode, imSrc->mode) || imDst->type != imSrc->type ||
-        imDst->bands != imSrc->bands || imDst->xsize != imSrc->xsize ||
```

Comment:
If the images are the same mode, then they will naturally have the same type and number of bands.